network/tailscale/tailscale.yaml

```diff
@@ -0,0 +1,73 @@
+name: tailscale
+depends:
+  - service: cri
+  - network:
+    - addresses
+    - connectivity
+    - etcfiles
+container:
+  entrypoint: ./containerboot
+  args: []
+  environment:
+  - PATH=/usr/local/bin:/sbin
+  - TS_SOCKET=/var/run/tailscale/tailscaled.sock
+  - TS_STATE_DIR=/var/lib/tailscale
+  - TS_USERSPACE=false
+  - TS_ACCEPT_DNS=true
+  security:
+    maskedPaths: []
+    readonlyPaths: []
+    writeableRootfs: true
+    writeableSysfs: true
+  mounts:
+  # libs
+  - source: /lib
+    destination: /lib
+    type: bind
+    options:
+      - bind
+      - ro
+  # more libs
+  - source: /usr/lib
+    destination: /usr/lib
+    type: bind
+    options:
+      - bind
+      - ro
+  - source: /sbin
+    destination: /sbin
+    type: bind
+    options:
+      - bind
+      - ro
+  - source: /usr/local/bin
+    destination: /usr/local/bin
+    type: bind
+    options:
+      - bind
+      - ro
+  - source: /dev/net/tun
+    destination: /dev/net/tun
+    type: bind
+    options:
+      - bind
+      - rw
+  - source: /var/lib/tailscale
+    destination: /var/lib/tailscale
+    type: bind
+    options:
+      - bind
+      - rw
+  - source: /var/run/tailscale
+    destination: /var/run/tailscale
+    type: bind
+    options:
+      - bind
+      - rw
+  - source: /etc/resolv.conf
```

Comment:
This is potentially an issue.

Upgrading starts a race condition, where this points to the Tailscale service, which isn't running as it's being upgraded.

I thought this was meant to be more a 'ram' version in talos, but it seems modifying the file persists. Was hoping the 'reboot' procedure will go back to normal DNS resolution until the Tailscale service starts.

Reply:
--staged seemed to have helped. Unsure exactly why, but I think that does the bulk of the upgrade *after* a reboot, which may has reset to normal dns.

Reply:
Removed magic dns for first pass, it probably needs split dns as a seperate service, and is too much a PITA right now.